src/apps/relay/netengine.c

```diff
@@ -1605,6 +1605,32 @@ void run_listener_server(struct listener_server *ls) {
       }
     }
 
+    // Check if we are draining
+    if (turn_params.drain_turn_server == DRAINMODE_REQUESTED) {
+      // Tell each turn_service we are draining
+      for (size_t i = 0; i < ls->services_number; i++) {
+        for (size_t j = 0; j < get_real_general_relay_servers_number(); j++) {
+          if (ls->udp_services[i] && ls->udp_services[i][j]) {
+            ls->udp_services[i][j]->ts->is_draining = 1;
+          }
+          if (ls->dtls_services[i] && ls->dtls_services[i][j]) {
+            ls->dtls_services[i][j]->ts->is_draining = 1;
+          }
+        }
+      }
+      for (size_t i = 0; i < turn_params.aux_servers_list.size; i++) {
+        for (int j = 0; j < get_real_general_relay_servers_number(); j++) {
+          if (ls->aux_udp_services[i] && ls->aux_udp_services[i][j]) {
+            ls->aux_udp_services[i][j]->ts->is_draining = 1;
+          }
```

Comment:
Don't we need to also drain TCP and TLS?

Reply:
Yes, I suppose we do, I will look into this as well.

Reply:
Actually I found a better way to do this, that means I don't need to move dtls_listener_relay_server_info to the header file.  Changes coming soon.